File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = [0] *2
    l = func(yellow)
    for i in l[::-1]:
        if (i[0]+2)*2 + i[1]*2 == brown:
            answer = [i[0]+2,i[1]+2]
            break
    return answer

def func(yellow):
    t = ()
    result = []
    for i in range(1,yellow+1):
        if yellow%i == 0:
            t = (i,yellow//i)
            result.append(t)
    return result

File: 프로그래머스/2/42842. 카펫/카펫.java
import java.util.*;

class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        int count = (brown - 4)/2;
        for(int i=1;i<=count/2;i++) {
            if (i*(count-i) == yellow) {
                int a = i + 2;
                int b = (count-i) + 2;
                answer[0] = Math.max(a,b);
                answer[1] = Math.min(a,b);
            }
        }
        return answer;
    }
}